ambry-api/src/main/java/com.github.ambry/account/Container.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+
+/**
+ * <p>
+ *   A representation of a container. A container virtually groups a number of blobs under the same {@link Account},
+ * so that an operation on a container can be applied to all the blobs of the container. When posting a blob, a
+ * container needs to be specified. There can be multiple containers under the the same {@link Account}, but containers
+ * cannot be nested.
+ * </p>
+ * <p>
+ *   Container name is provided by a user as an external reference to that container. Container id is an internal
+ *   identifier of the container, and is one-to-one mapped to a container name. Container name/id has to be distinct
+ *   within the same {@link Account}, but can be the same across different {@link Account}s.
+ * </p>
+ * <p>
+ *   Container is serialized into {@link JSONObject} in the highest metadata version, which is version 1 for now.
+ *   Below lists all the metadata versions and their formats:
+ * </p>
+ *  <pre><code>
+ *  {
+ *    "containerName": "MyPrivateContainer",
+ *    "description": "This is my private container",
+ *    "isPrivate": "true",
+ *    "containerId": 0,
+ *    "version": 1,
+ *    "status": "active"
+ *    "parentAccountId": "101"
+ *  }
+ *  </code></pre>
+ *  <p>
+ *    A container object is immutable. To update a container, refer to {@link ContainerBuilder} and
+ *  {@link AccountBuilder}.
+ *  </p>
+ */
+public class Container {
+  // static variables
+  static final String CONTAINER_METADATA_VERSION_KEY = "version";
+  static final String CONTAINER_NAME_KEY = "containerName";
+  static final String CONTAINER_ID_KEY = "containerId";
+  static final String CONTAINER_STATUS_KEY = "status";
+  static final String CONTAINER_DESCRIPTION_KEY = "description";
+  static final String CONTAINER_IS_PRIVATE_KEY = "isPrivate";
+  static final String CONTAINER_PARENT_ACCOUNT_ID_KEY = "parentAccountId";
+  static final short CONTAINER_METADATA_VERSION_1 = 1;
+  static final short HIGHEST_CONTAINER_METADATA_VERSION = CONTAINER_METADATA_VERSION_1;
+  // container field variables
+  private final Short id;
+  private final String name;
+  private final ContainerStatus status;
+  private final String description;
+  private final Boolean isPrivate;
```

Comment:
Would prefer to have a single place to check all the required fields, and `name` fields will anyway be checked in the constructors.